web/packages/extension/src/player.ts

```diff
@@ -241,28 +243,32 @@ async function loadSwf(swfUrl: string) {
     });
 }
 
-window.addEventListener("pageshow", async () => {
+async function loadSwfFromHash() {
     const url = new URL(window.location.href);
     // Hash always starts with #, gotta slice that off
     const swfUrl = url.hash.length > 1 ? url.hash.slice(1) : null;
     if (swfUrl) {
+        webURL.value = swfUrl;
         await loadSwf(swfUrl);
     }
+}
+
+window.addEventListener("pageshow", async () => {
+    await loadSwfFromHash();
+});
+
+window.addEventListener("hashchange", async () => {
+    await loadSwfFromHash();
 });
 
 window.addEventListener("DOMContentLoaded", async () => {
-    const webFormSubmit = document.getElementById(
-        "web-form-submit",
-    ) as HTMLButtonElement;
-    if (webFormSubmit) {
-        webFormSubmit.addEventListener("click", function () {
-            const webURL = document.getElementById(
-                "web-url",
-            ) as HTMLInputElement;
-            if ((webURL?.value || "") !== "") {
-                loadSwf(webURL.value);
-                window.location.href = "#" + webURL.value;
-            }
-        });
-    }
+    webFormSubmit.addEventListener("click", function () {
+        if (webURL.value !== "") {
+            loadSwf(webURL.value);
+            window.location.href = "#" + webURL.value;
```

Comment:
I don't know if there's any strict difference between them, but `window.location.hash = webURL.value` does the same thing a bit more directly.
Not a big deal though!

Reply:
Will change to `hash`. It also occurs to me, the line before this, `loadSwf(webURL.value);`, shouldn't be necessary with this firing the `hashchange` event which already does that.

Reply:
The exceptions is that now if you don't change the URL before clicking "Load", the animation won't reload, but it's probably better to rely on the reload icon for reloading the swf anyway, as that gives a warning confirmation.